src/EvoSC.Manialinks/Interfaces/IManialinkManager.cs

```diff
@@ -0,0 +1,184 @@
+﻿using EvoSC.Common.Interfaces.Models;
+using EvoSC.Manialinks.Interfaces.Models;
+
+namespace EvoSC.Manialinks.Interfaces;
+
+/// <summary>
+/// Manialink template & ManiaScript registry, and can render, display and hide Manialink to players.
+/// </summary>
+public interface IManialinkManager
+{
+    /// <summary>
+    /// Add all the default templates from EvoSC.
+    /// </summary>
+    /// <returns></returns>
+    public Task AddDefaultTemplatesAsync();
+    
+    /// <summary>
+    /// Register a template, does not pre-process the template.
+    /// </summary>
+    /// <param name="template">The template to add.</param>
+    internal void AddTemplate(IManialinkTemplateInfo template);
```

Comment:
Adding internal methods in the interface creates a problem for tests of other projects as the IManialinkManager cannot be mocked. A cleaner solution would be to create an additional internal interface that contains the internal methods which extends from IManialinkManager and inherits all public method definitions. This way the Evosc.Modules project can use the new interface and all other projects can use the IManialinkManager interface and mock them in their tests.

Reply:
But how would you set up the service container for this then?

Reply:
I tried it out. There are a few points to note though. The interface cannot be internal and has to be public (it can have the internal methods though). 

```csharp
namespace EvoSC.Manialinks.Interfaces;

/// <summary>
/// Manialink template & ManiaScript registry, and can render, display and hide Manialink to players.
/// </summary>
public interface IInternalManialinkManager: IManialinkManager
{
    /// <summary>
    /// Register a template, does not pre-process the template.
    /// </summary>
    /// <param name="template">The template to add.</param>
    internal void AddTemplate(IManialinkTemplateInfo template);
    
    /// <summary>
    /// Register a new ManiaScript resource.
    /// </summary>
    /// <param name="maniaScript">The ManiaScript to add</param>
    internal void AddManiaScript(IManiaScriptInfo maniaScript);
}
```

```csharp
public interface IManialinkManager
{
    /// <summary>
    /// Add all the default templates from EvoSC.
    /// </summary>
    /// <returns></returns>
    public Task AddDefaultTemplatesAsync();
    
    // All other public methods ...
}
````

```csharp
public class ManialinkManager : IInternalManialinkManager
{
    // same as before ....
}
```
Register the singleton like that. SimpleInjector makes sure they are using the same instance. 
```csharp
services.RegisterSingleton<IManialinkManager, ManialinkManager>();
services.RegisterSingleton<IInternalManialinkManager, ManialinkManager>();
````

`object.ReferenceEquals(_services.GetInstance<IManialinkManager>(), _services.GetInstance<IInternalManialinkManager>())` returns true

We should probably use a different name though.

Reply:
If you want I could add the commit this branch to add this,